content/developer/tutorials/data_modules.rst

```diff
@@ -0,0 +1,906 @@
+    </header>
+
+It is also possible to add an entry in the 'cog menu' to run this action (e.g. to avoid adding
+buttons to view that are already crowded). To do so, you can _bind_ your server action to the model
+and to specific types of views:
+
+.. code-block:: xml
+
+    <record model="ir.actions.server" id="action_x_estate_property_refuse_all_offers">
+        <field name="name">Refuse all offers</field>
+        <field name="model_id" ref="estate.model_real_estate_property"/>
+        <field name="state">code</field>
+        <field name="binding_model_id" ref="estate.model_real_estate_property"/>
+        <field name="binding_view_types">tree,form</field>
+        <field name="code"><![CDATA[
+    for property in records:
+        property.x_offer_ids.write({'x_status': 'refused'})
+        ]]></field>
+    </record>
+
+This will make the action available in the cog menu of the ``x_estate.property`` model,
+in the list views (when one or more records are selected via the checkbox) and form views.
+
+.. exercise:: Add an action to the Real Estate Offer model.
+
+    - Add a server action to the ``x_estate.property.offer`` model that sets the ``x_status``
+      field of an offer to ``Accepted`` and update the selling price and buyer of the property
+      to which the offer is attached accordingly. This action also marks all the other offers
+      on the same property as ``Refused``.
+    - Include a button in the embedded list view of offers that allows to execute this action
+
+    .. image:: data_modules/offer_accept_button.png
+       :align: center
+
+Overriding Python Models
+------------------------
+
+Via UI Elements
+~~~~~~~~~~~~~~~
+
+Unlike in Python modules, it is not possible to override a Python model's method cleanly.
+
+However, it is possible (in some cases) to replace the elements of the UI that call
+these methods and to intercept the calls to these methods in a server action.
+
+A typical example would be an integration with the ``Sales`` app of Odoo. Let's imagine that your
+Real Estate module integrates with the Sales application so that when a specific product is sold
+(e.g. a quote for managing the sale of a property), you want to automatically create a new property
+record in your module.
+
+To achieve this, you will need to:
+
+- create a server action that calls the original method of the button and add custom logic before
+  or after that method call
+- replace the button in the view with a custom button that calls the server action
+
+
+.. code-block:: xml
+
+    <record model="ir.ui.view" id="view_sale_order_form">
+        <field name="name">sale.order.form.inherit.estate</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='action_confirm'][@type='object']" position="attributes">
```

Comment:
Is the type really needed?
```suggestion
            <xpath expr="//button[@name='action_confirm']" position="attributes">
```

Reply:
no for the first one, but it is for the second one otherwise you just replace the button you define above. i thought it'd be clearer to have both xpath be similar, since in any case it's not incorrect and specifity in xpath is not necessarily a bad thing